File: segmentSupport/src/main/java/com/clumob/segment/support/pager/recycler/SegmentTabsHelper.java
package com.clumob.segment.support.pager.recycler;

import com.google.android.material.tabs.TabLayout;
import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.SnapHelper;

/**
 * Created by prashant.rathore on 13/07/18.
 */

public class SegmentTabsHelper {

    public static void attach(final RecyclerView recyclerView,
                              final TabLayout tabLayout,
                              SnapHelper snapHelper) {

        SegmentPageChangeHelper segmentPageChangeHelper = new SegmentPageChangeHelper(snapHelper);
        segmentPageChangeHelper.attachRecyclerView(recyclerView);
        attach(tabLayout, segmentPageChangeHelper);
    }

    public static void attach(final TabLayout tabLayout, final SegmentPageChangeHelper pageChangeHelper) {
        tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                pageChangeHelper.scrollToPosition(tab.getPosition());
            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) {

            }

            @Override
            public void onTabReselected(TabLayout.Tab tab) {

            }
        });
        pageChangeHelper.addPageChangeListner(new SegmentPageChangeHelper.PageChangeListner() {
            @Override
            public void onPageChanged(final int position) {
                tabLayout.getTabAt(position).select();
            }

            @Override
            public void onPageScrolled(int position, float offset) {
                tabLayout.setScrollPosition(position, offset, true);
            }
        });
    }

}

File: segmentSupport/src/main/java/com/clumob/segment/support/pager/recycler/SegmentTabsHelper.kt
//package com.clumob.segment.support.pager.recycler
//
//import androidx.recyclerview.widget.RecyclerView
//import androidx.recyclerview.widget.SnapHelper
//import com.clumob.segment.support.pager.recycler.SegmentPageChangeHelper.PageChangeListner
//import com.google.android.material.tabs.TabLayout
//import com.google.android.material.tabs.TabLayout.OnTabSelectedListener
//
///**
// * Created by prashant.rathore on 13/07/18.
// */
//object SegmentTabsHelper {
//    fun attach(recyclerView: RecyclerView,
//               tabLayout: TabLayout,
//               snapHelper: SnapHelper) {
//        val segmentPageChangeHelper = SegmentPageChangeHelper(snapHelper)
//        segmentPageChangeHelper.attachRecyclerView(recyclerView)
//        attach(tabLayout, segmentPageChangeHelper)
//    }
//
//    fun attach(tabLayout: TabLayout, pageChangeHelper: SegmentPageChangeHelper) {
//        tabLayout.addOnTabSelectedListener(object : OnTabSelectedListener {
//            override fun onTabSelected(tab: TabLayout.Tab) {
//                pageChangeHelper.scrollToPosition(tab.position)
//            }
//
//            override fun onTabUnselected(tab: TabLayout.Tab) {}
//            override fun onTabReselected(tab: TabLayout.Tab) {}
//        })
//        pageChangeHelper.addPageChangeListner(object : PageChangeListner {
//            override fun onPageChanged(position: Int) {
//                tabLayout.getTabAt(position)!!.select()
//            }
//
//            override fun onPageScrolled(position: Int, offset: Float) {
//                tabLayout.setScrollPosition(position, offset, true)
//            }
//        })
//    }
//}